filers/default/src/main/java/org/commonjava/indy/filer/def/DefaultGalleyStorageProvider.java

```diff
@@ -292,4 +300,17 @@ public synchronized CacheProvider getCacheProvider()
 
         return cacheProvider;
     }
+
+    @Produces
+    @Default
+    @ApplicationScoped
+    public synchronized SpecialPathManager getSpecialPathManager()
```

Comment:
Still not sure I understand this...is it just to allow this provider class to initialize the CacheProvider on the SpecialPathManager instance?

Reply:
As we will use some customized instance of this SpecialPathManager to inject some CacheProvider(or PathGenerator whatever), and we did not mark this SpecialPathManagerImpl as a CDI bean, that means we need to construct it in indy all by ourselves to inject extra fields, and provides this producer.  That's why I add @Alternative in galley SpecialPathManagerImpl because we use this producer as @Default.